tests/test_daemon_query.py

```
@@ -0,0 +1,284 @@
+                    main()
+                except SystemExit:
+                    pass
+
+            # Verify query_daemon was called with JSON payload, not {"cmd": "ping"}
+            mock_query_func.assert_called_once()
+            called_with_project, called_with_command = mock_query_func.call_args[0]
+            assert called_with_project == project_path
+            assert called_with_command == json.loads(json_payload)
+            assert called_with_command != {"cmd": "ping"}
+
+    def test_query_complex_json_payload(self, capsys):
```

Ruff (0.14.11)

Unused method argument: `capsys`

(ARG002)